lib/src/runner/reporter/xunit.dart

```diff
@@ -0,0 +1,363 @@
+  int beginningTime;
+  int endTime;
+  Map error = {};
+  String name;
+  String path;
+  bool skipped = false;
+  String skipReason;
+
+  XunitTestResult(this.name, this.path, this.beginningTime);
+}
+
+class XunitTestSuite {
+  int errored = 0;
+  int failed = 0;
+  int skipped = 0;
+  List testResults = [];
+  int tests = 0;
+
+  add(XunitTestResult test) {
+    testResults.add(test);
+  }
+}
+
+/// A reporter that returns xunit compatible output.
+
+class XunitReporter implements Reporter {
+  /// The engine used to run the tests.
+  final Engine _engine;
+
+  /// A stopwatch that tracks the duration of the full run.
+  final _stopwatch = new Stopwatch();
+
+  /// Whether we've started [_stopwatch].
+  ///
+  /// We can't just use `_stopwatch.isRunning` because the stopwatch is stopped
+  /// when the reporter is paused.
+  var _stopwatchStarted = false;
+
+  /// Whether the reporter is paused.
+  var _paused = false;
+
+  /// The set of all subscriptions to various streams.
+  final _subscriptions = new Set<StreamSubscription>();
+
+  /// Record of all tests run.
+  Map _testCases = {};
+
+  /// Number of failures that occur during the test run.
+  int _failureCount = 0;
+
+  /// Number of errors that occur during the test run.
+  int _errorCount = 0;
+
+  /// Map containing Xunit hierarchy.
+  Map _groupStructure = {};
+
+  /// Watches the tests run by [engine] and records its results.
+  static XunitReporter watch(Engine engine, {bool verboseTrace: false}) {
+    return new XunitReporter._(engine, verboseTrace: verboseTrace);
+  }
+
+  XunitReporter._(this._engine, {bool verboseTrace: false}) {
+    _subscriptions.add(_engine.onTestStarted.listen(_onTestStarted));
+
+    /// Convert the future to a stream so that the subscription can be paused or
+    /// canceled.
+    _subscriptions.add(_engine.success.asStream().listen(_onDone));
+  }
+
+  void pause() {
+    if (_paused) return;
+    _paused = true;
+
+    _stopwatch.stop();
+
+    for (var subscription in _subscriptions) {
+      subscription.pause();
+    }
+  }
+
+  void resume() {
+    if (!_paused) return;
+    if (_stopwatchStarted) _stopwatch.start();
+
+    for (var subscription in _subscriptions) {
+      subscription.resume();
+    }
+  }
+
+  void cancel() {
+    for (var subscription in _subscriptions) {
+      subscription.cancel();
+    }
+    _subscriptions.clear();
+  }
+
+  /// A callback called when the engine begins running [liveTest].
+  void _onTestStarted(LiveTest liveTest) {
+    if (liveTest.suite is! LoadSuite) {
+      if (!_stopwatch.isRunning) _stopwatch.start();
+
+      // The engine surfaces load tests when there are no other tests running,
+      // but because the expanded reporter's output is always visible, we don't
+      // emit information about them unless they fail.
+      _subscriptions.add(liveTest.onStateChange
+          .listen((state) => _onStateChange(liveTest, state)));
+    } else if (_engine.active.length == 1 &&
```

Comment:
since there's no body within this statement, just kill the whole `else if` block